### PR TITLE
Modular computers' UI overlay variable addition

### PR DESCRIPTION
**Pull request by Stalkeros2**
## About The Pull Request
Adds a new variable icon_overlays that, if set, replaces the modular computers' overlays with the ones from a different file; if not, defaults to the initial icon file's overlays.
Readjusts some of the update_overlays proc code to better accommodate the change.
## Why It's Good For The Game
Makes it easier for admins, players and downstreams (see: removes the sheer amount of needless copypasted screen states like the image below) to implement custom modular computers.
image
Like, really, this is horrible. :(
## Changelog
:cl: Stalkeros
code: Modular computer overlays' can now be properly edited.
/:cl:

### CODE REVIEW

File: code/modules/modular_computers/computers/item/computer.dm [hunk 1]
```diff
@@ -392,15 +394,13 @@
 
 /obj/item/modular_computer/update_overlays()
 	. = ..()
-	var/init_icon = initial(icon)
-	if(!init_icon)
-		return
+	var/ui_overlay = icon_overlays ? icon_overlays : initial(icon)
```
Comment (JohnFulpWillard) [inline code/modules/modular_computers/computers/item/computer.dm:397]: This should be a var on programs instead of tablets

Comment (JohnFulpWillard) [reply code/modules/modular_computers/computers/item/computer.dm:397]: Actually no, I suppose this is fine. Overwriting the icon from the program would put the icon of tablets as the icon for computers and laptops.

File: code/modules/modular_computers/computers/item/pda.dm [hunk 2]
```diff
@@ -67,14 +67,14 @@
 		var/datum/computer_file/program/program_type = new programs
 		store_file(program_type)
 
-/obj/item/modular_computer/pda/update_overlays()
+/obj/item/modular_computer/pda/update_overlays(ui_overlay)
```
Comment (MrMelbert) [inline code/modules/modular_computers/computers/item/pda.dm:70]: What's the point of this parameter exactly? Is this proc called somewhere where it passes this?

I don't really think you should just add parameters to this, in any case

File: code/modules/modular_computers/computers/item/computer.dm (hunk 1 above)
Comment (MrMelbert) [inline code/modules/modular_computers/computers/item/computer.dm:397]:
```suggestion
	var/ui_overlay = icon_overlays || initial(icon)
```

**Review by JohnFulpWillard — COMMENTED** [review]
Actually thinking about it- whats the point of this?
Downstreams can easily just change the icon. This doesn't cut down on copy pasted icons at all, this really just adds another icon var for no real reason. The only difference would be the icon of the pda itself, which is nothing compared to the point of the dmi to be holding the sprites of the programs (and they should be split anyways).

**Comment by Stalkeros2** [comment]
> Downstreams can easily just change the icon; This doesn't cut down on copy pasted icons at all, this really just adds another icon var for no real reason.

Incorrect, because currently the icon file calls for the initial file (itself, duh) for all of its overlays. As such, without this variable, any overlays would have to be copypasted into the new PDA file to actually properly appear, increasing the file size for no reason whatsoever and generally being a bad practice.

**Comment by JohnFulpWillard** [comment]
> Incorrect, because currently the icon file calls for the initial file (itself, duh) for all of its overlays. As such, without this variable, any overlays would have to be copypasted into the new PDA file to actually properly appear, increasing the file size for no reason whatsoever and generally being a bad practice.

With this PR, any downstreams will still have to copy paste all overlays for all programs if they plan to add more or resprite existing ones, the only difference is that they won't need the sprites to the PDA itself, which again should be separate icons anyways.

**Comment by JohnFulpWillard** [comment]
I still want to see this PR happen, I just think that it is useless to do in the current state.
Split PDA/Laptops/Computers into their own dmi file that is read by the icon, and the program overlays are from the new var you've added, and I'll be glad.

**Comment by Stalkeros2** [comment]
Been a bit busy; gonna do this rn.

**Comment by san7890** [comment]
Please feel free to do the requested changes in this PR in a brand-new PR so it doesn't keep languishing at the bottom of the PR tracker for three weeks and double-stales with no progress. It's alright if you're busy with life or whatever, just want to keep the tracker clean (and it shouldn't be too much of an issue to make a new PR).

**Pull request closed without merging**